debian/logitechmediaserver.service

```diff
@@ -122,6 +122,12 @@ User=squeezeboxserver
 LogsDirectory=squeezeboxserver
 LogsDirectoryMode=0755
 
+# Since we launch LMS in the foreground (ie not daemonized) logs are
```

Comment:
I'm not familiar with the details of systemd. But why would we be launching in the foreground?

Reply:
The reason for launching it in the foreground is because systemd prefers that:
https://github.com/Logitech/slimserver-platforms/pull/41/files#diff-6b0ed038e21a225a2b300b7559c57b850c5e87e0ed36bddb2a27b6c0f265bad2L139-L145

.. but it's also possible to set `Type=forking` and daemonize LMS. That would also solve the problem with duplicate logs.

I have no strong opinion on which way to go. @mw9, since you wrote the systemd service unit, do you have any suggestions here?

Reply:
Yes, I have a strong suggestion: use LMS' `--quiet` option at start up. This is what it is designed for.

Do this by adding `--quiet` to `SLIMOPTIONS` in file `/etc/default/logitechmediaserver`.

Simple example for the default case: `SLIMOPTIONS="--quiet"`

Perhaps this should be made standard behaviour.

For what it's worth, this option has been present in LMS since time immemorial, but it Never Actually Worked until a couple of years ago. But it does now work, as a result of this PR: https://github.com/Logitech/slimserver/pull/696

I'd guess that the PR was made to deal with this precise problem.

As to what's going on under the good: by default LMS's logging system logs to `STDOUT` as well as the usual log file. `--quiet` suppresses this behaviour (at least for the last two years).

It was not an issue before the systemd service unit was put in place, because LMS was started by `/usr/sbin/squeezeboxserver_safe`, and that script just redirects `STDOUT` & `STDERR` to `/dev/null`.

So, I think the action is to leave the systemd unit file alone, and perhaps add `--quiet` to the default install.

Reply:
Thanks for the input. I'm fine with using `--quiet` also. One possible "problem" with `--quiet` could be that there are a few places where logs are dropped also from the LMS log files if quiet is set:
* https://github.com/Logitech/slimserver/blob/public/8.4/Slim/Schema.pm#L1770
* https://github.com/Logitech/slimserver/blob/public/8.4/Slim/Music/Info.pm#L198

.. but it may not be a big problem.

> It was not an issue before the systemd service unit was put in place

Then I would prefer to put the fix in systemd's unit file where the cause of the problem is. If the fix is put in `/etc/config/logitechmediaserver` it's redundant if systemd isn't used and if systemd is used it's not easy to understand that there's a dependency/connection between the LMS configuration file and systemd's unit file. Just my thought, I'm not going to fight for this :-)

Reply:
@robho 

There is a problem with redirecting `STDOUT` & `STDERR` to `/dev/null`. One loses the ability to see any output from LMS should it fail to start, and that does, sometimes, happen.

See, for example, this issue: _Startup errors of LMS get obscured by squeezeboxserver_safe script_ https://github.com/Logitech/slimserver-platforms/issues/6.

Hence my deliberate decision, when drafting the systemd unit file, **NOT** to throw `STDOUT` & `STDERR` away. Your proposed change would reintroduce this issue. I can attest that the problem is real.

> Then I would prefer to put the fix in systemd's unit file where the cause of the problem is.

My view is that the cause of the problem is/was in the `squeezeboxserver_safe` script, not the systemd unit file. It simply should not have thrown away `STDOUT` & `STDERR` but used `--quiet` instead. But `--quiet` was ineffective...

> One possible "problem" with --quiet could be that there are a few places where logs are dropped also from the LMS log files if quiet is set: .. but it may not be a big problem.

I wasn't aware of that. I agree it is probably not a big problem, perhaps a more thorough review is wanted.

> ...and if systemd is used it's not easy to understand that there's a dependency/connection between the LMS configuration file and systemd's unit file

The systemd unit file "depends on" the LMS configuration file (`/etc/default/logitechmediaserver`) by design. This is to maintain compatibility between debian based systemd installations and debian based "no systemd here" installations. For example, devuan.

There are some notes in the systemd unit file that seek to address the point. Perhaps they are not clear enough.
https://github.com/Logitech/slimserver-platforms/blob/public/8.4/debian/logitechmediaserver.service#L55

> Just my thought, I'm not going to fight for this :-)

I hope the above notes indicate why I feel slightly more strongly about the point. But I think a PR that made `--quiet` the default start up option would be very welcome, if you can see that as an acceptable solution. I agree that the  issue you have raised in this PR is real, and should be dealt with. I just never got round to it.

Reply:
I created #42 where --quiet is used.

Reply:
#42 looks good to me.